File: RecyclerViewJavaLessons(17)/RecyclerViewCustomMultitypeAdapterHeaderAndFooter_3/app/src/main/java/com/example/recyclerviewcustommultitypeadapter/activity/MainActivity.java
package com.example.recyclerviewcustommultitypeadapter.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;

import com.example.recyclerviewcustommultitypeadapter.Member.Member;
import com.example.recyclerviewcustommultitypeadapter.R;
import com.example.recyclerviewcustommultitypeadapter.adapter.CustomAdapter;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private Context context;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();

        List<Member> members = prepareMemberList();
        refreshAdapter(members);

    }

    private void initViews() {
        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context,1));
    }

    private void refreshAdapter(List<Member> members){
        CustomAdapter adapter = new CustomAdapter(context,members);
        recyclerView.setAdapter(adapter);
    }

    List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        members.add(new Member());
        for (int i = 0; i < 20; i++) {
            if (i==0 || i==5 || i == 10 || i == 15){
                members.add(new Member("Feruz", "Davlatov",false));
            } else {
                members.add(new Member("Feruz", "Davlatov",true));
            }
        }
        members.add(new Member());
        return members;
    }

}

File: RecyclerViewKotlinLessons(17)/RecyclerViewCustomMultitypeAdapter_2/app/src/main/java/com/example/recyclerviewcustommultitypeadapter_2/activity/MainActivity.kt
package com.example.recyclerviewcustommultitypeadapter_2.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerviewcustommultitypeadapter_2.R
import com.example.recyclerviewcustommultitypeadapter_2.adapter.CustomAdapter
import java.lang.reflect.Member
import java.util.ArrayList

class MainActivity : AppCompatActivity() {

    private lateinit var context: Context
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        val member = prepareMemberList()
        refreshAdapter(member)
    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

    private fun refreshAdapter(members: List<com.example.recyclerviewcustommultitypeadapter_2.model.Member>) {
        val adapter = CustomAdapter(context, members)
        recyclerView.adapter = adapter
    }

    private fun prepareMemberList(): ArrayList<com.example.recyclerviewcustommultitypeadapter_2.model.Member> {
        val members = ArrayList<com.example.recyclerviewcustommultitypeadapter_2.model.Member>()
        for (i in 0..20){
            if (i == 0 || i == 5 || i == 10 || i == 15) {
                members.add(com.example.recyclerviewcustommultitypeadapter_2.model.Member("Feruz", "Davlatov", true))
            } else {
                members.add(com.example.recyclerviewcustommultitypeadapter_2.model.Member("Feruz", "Davlatov", false))
            }
        }
        return members
    }
}